apps/emqx_postgresql/src/emqx_postgresql.erl

```diff
@@ -148,13 +153,137 @@ on_start(
             {error, Reason}
     end.
 
-on_stop(InstId, _State) ->
+on_stop(InstId, State) ->
     ?SLOG(info, #{
         msg => "stopping_postgresql_connector",
         connector => InstId
     }),
+    close_connections(State),
     emqx_resource_pool:stop(InstId).
 
+close_connections(#{pool_name := PoolName} = _State) ->
+    WorkerPids = [Worker || {_WorkerName, Worker} <- ecpool:workers(PoolName)],
+    close_connections_with_worker_pids(WorkerPids),
```

Comment:
Nit: if `WorkerPids = []`, it won't close anything.

Reply:
If there are no worker pids then there are no connections to close.